### PR TITLE
Add option for encoding

### PR DESCRIPTION
**Pull request by janbrohl**
See #13

### CODE REVIEW

File: sphinxcontrib/datatemplates/directive.py [hunk 1]
```diff
@@ -26,39 +27,67 @@ class DataTemplate(rst.Directive):
...
+        try:
+            if encoding is None:
+                f = open(filename, 'r')
+            else:
+                f = codecs.open(filename, 'r', encoding=encoding)
+            return yaml.safe_load(f)
+        finally:
+            f.close()
+
+    def _load_data(self, env, data_source, encoding):
```
Comment (dhellmann) [inline sphinxcontrib/datatemplates/directive.py:195]: We could have 1 block in _load_data() to manage opening the file, and then pass a file handle to the other load functions. That would cut down on the number of places we have to pay attention to the encoding.

File: sphinxcontrib/datatemplates/directive.py [hunk 2]
```diff
@@ -26,39 +27,67 @@ class DataTemplate(rst.Directive):
...
-                        r = csv.DictReader(f, dialect=dialect)
-                else:
-                    if dialect is None:
-                        r = csv.reader(f)
-                    else:
-                        r = csv.reader(f, dialect=dialect)
-                return list(r)
+            return self._load_csv(filename, encoding)
         elif "xml" in mimetypes.guess_type(data_source)[0]:
+            # there are many XML based formats
```
Comment (dhellmann) [inline sphinxcontrib/datatemplates/directive.py:204]: Although I notice there isn't a separate local load function for XML. Does the XML parser deal with encoding itself?

Comment (janbrohl) [reply sphinxcontrib/datatemplates/directive.py:204]: Exactly. XML files state their encoding at their beginning or must have it autodetectable (See https://www.w3.org/TR/2006/REC-xml11-20060816/#charencoding ).
It is not actually stated in the docs but [xml.etree.ElementTree.parse](https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.parse) seems to do autodetection.

File: sphinxcontrib/datatemplates/directive.py (hunk 1 above)
Comment (janbrohl) [reply sphinxcontrib/datatemplates/directive.py:195]: Actually I think so, too - but open for CSV does take different parameters and for XML there is no open at all.

Comment (janbrohl) [reply sphinxcontrib/datatemplates/directive.py:195]: Maybe it's better to tackle #17 first.

Comment (dhellmann) [reply sphinxcontrib/datatemplates/directive.py:195]: Yeah, for #17 I could see refactoring the data loading to a method of the subclass and then it wouldn't feel as repetitive.

File: sphinxcontrib/datatemplates/directive.py (hunk 2 above)
Comment (dhellmann) [reply sphinxcontrib/datatemplates/directive.py:204]: Yeah, that's what I thought I remembered. It has been a while since I've done any serious XML.

**Review by dhellmann — APPROVED** [review]

**Pull request merged**